File: EJERCICIO_1/EJERCICIOSPHP/bonus.php
<?php


abstract class Bonus {
    protected $percentage; // Porcentaje de bonificación
    protected $days;       // Días trabajados
    protected $valueDay;  

    public function __construct(SalaryBass $salaryBass, $percentage, $days) {
        $this->valueDay = $salaryBass->getValueDay(); // Obtener el valor diario desde SalaryBass
        $this->percentage = $percentage;
        $this->days = $days;
    }

    abstract public function calculateBonus();

    public static function calculateTotalBonus($dayTime, $nightTime, $holiday) {
        return $dayTime->calculateBonus() + $nightTime->calculateBonus() + $holiday->calculateBonus();
    }

}




?>

File: EJERCICIO_1/EJERCICIOJAVA/bonus.java
package EJERCICIOJAVA;

public  abstract class bonus {
    protected double percentage; 
    protected int days;          
    protected double valueDay;  

    public bonus(salaryBass salary, double percentage, int days) {
        this.valueDay = salary.getValueDay(); 
        this.percentage = percentage;
        this.days = days;
    }

    public  abstract double calculateBonus()
     ;

    public static double calculateTotalBonus(dayTime dayTimeBonus, nightTime nightTimeBonus, holiDay holidayBonus) {
        return  dayTimeBonus.calculateBonus() + nightTimeBonus.calculateBonus() + holidayBonus.calculateBonus();
    }

    }
